FWCore/Framework/src/EventProcessor.cc

```diff
@@ -486,13 +483,13 @@ namespace edm {
     ScheduleItems items;
 
     //initialize the services
-    std::shared_ptr<std::vector<ParameterSet> > pServiceSets = processDesc->getServicesPSets();
-    ServiceToken token = items.initServices(*pServiceSets, *parameterSet, iToken, iLegacy, true);
+    auto pServiceSets = processDesc->getServicesPSets();
```

Comment:
I do not believe you are correct. The signature for the member function is
```
auto const& getServicesPSets() const {return services_;}
```
Therefore it can not do a move construction and must do a copy.